lib/UI/calendar/widgets/date_widget.dart

```diff
@@ -3,96 +3,97 @@ import 'package:exerlog/Bloc/workout_bloc.dart';
 import 'package:exerlog/Models/workout.dart';
 import 'package:exerlog/UI/global.dart';
 import 'package:exerlog/UI/prev_workout/prev_workout_page.dart';
-import 'package:exerlog/src/widgets/gradient_button.dart';
 import 'package:flutter/material.dart';
 
-class DateWidget extends StatefulWidget {
+class DateWidget extends StatelessWidget {
   final DateTime date;
 
   DateWidget(this.date);
 
-  @override
-  _DateWidgetState createState() => _DateWidgetState();
-}
-
-class _DateWidgetState extends State<DateWidget> {
   @override
   Widget build(BuildContext context) {
-    DateTime after = new DateTime(
-        widget.date.year, widget.date.month, widget.date.day, 23, 59, 59, 0, 0);
-    DateTime before = new DateTime(
-        widget.date.year, widget.date.month, widget.date.day, 0, 0, 0, 0, 0);
+    DateTime after = DateTime(
+      date.year,
+      date.month,
+      date.day,
+      23,
+      59,
+      59,
+      0,
+      0,
+    );
+    DateTime before = DateTime(
+      date.year,
+      date.month,
+      date.day,
+      0,
+      0,
+      0,
+      0,
+      0,
+    );
     return StreamBuilder<QuerySnapshot<Map<String, dynamic>>>(
       stream: getWorkoutOnDate(after, before),
       builder: (context, snapshot) {
-        if (snapshot.connectionState == ConnectionState.waiting) {
-          return Center(
-            child: CircularProgressIndicator(),
+        if (snapshot.hasError) {
+          return _buildDayContainer();
```

Comment:
The same thing is returned here as the final else statement. So this is unnecessary, just put a if statement to check if the snapshot does not have data and then handle the rest from there.

```
if (!snapshot.hasData) return _buildDaysContainer();
// The rest of the logic if the snapshot does have data;
```

Reply:
Done